sbncode/FlashMatch/FlashPredict_module.cc

```diff
@@ -247,78 +252,97 @@ void FlashPredict::produce(art::Event& evt)
   }
 
   // load OpHits previously created
-  art::Handle<std::vector<recob::OpHit>> ophits_h;
-  evt.getByLabel(fOpHitProducer, ophits_h);
-  if(!ophits_h.isValid()) {
-    mf::LogError("FlashPredict")
-      << "No optical hits from producer module "
-      << fOpHitProducer;
-    bk.nonvalidophit++;
-    updateBookKeeping();
-    for(size_t pId=0; pId<pfps_h->size(); pId++) {
-      if(!pfps_h->at(pId).IsPrimary()) continue;
-      const art::Ptr<recob::PFParticle> pfp_ptr(pfps_h, pId);
-      sFM_v->emplace_back(sFM(kNoScr, kNoScrTime, Charge(kNoScrQ),
-                              Flash(kNoScrPE), Score(kNoOpHInEvt)));
-      util::CreateAssn(*this, evt, *sFM_v, pfp_ptr, *pfp_sFM_assn_v);
+  std::vector<SimpleFlash> simpleFlashes;
```

Comment:
Starting from here and down to line 340 or somewhere around there you should encase this whole thing into a function whose output is the `std::vector<FlashMetrics> flashMetrics`. 

It would be neater and would save from having variables alive that are no longer necessary once `flashMetrics` is created.

Reply:
For this new function that I suggest make sure to assign the right scope to variables (for instance the `simpleFlashes` vector should only exist inside the `fIsSimple` if).